quill-ndbc/src/main/scala/io/getquill/context/ndbc/NdbcContext.scala

```diff
@@ -55,13 +57,22 @@ abstract class NdbcContext[I <: SqlIdiom, N <: NamingStrategy, P <: PreparedStat
   protected def withDataSource[T](f: DataSource[P, R] => T): Result[T] = wrap(f(dataSource))
 
   def transaction[T](f: => Future[T]): Future[T] = {
-    withDataSource { ds =>
-      ds.transactional(new Supplier[io.trane.future.Future[T]] {
-        override def get = f.toJava
-      })
-    }.flatMap(_.toScala)
+    val pool = FuturePool.apply(Executors.newCachedThreadPool())
```

Comment:
Say you do something like:
```scala
transaction {
  transaction {
    ... // And N more transaction { ... blocks
      transaction {
        query[Person]
      }.filter(p => ...)
    ... // and '}' N times
  }.filter(p => ...)
}
```
I am assuming that `newCachedThreadPool` "will reuse previously constructed threads when they are available" ([here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newCachedThreadPool())). So the above should be fine.